daemon/algod/api/server/v2/utils.go

```diff
@@ -178,3 +217,52 @@ func decode(handle codec.Handle, data []byte, v interface{}) error {
 	}
 	return nil
 }
+
+// Helper to convert basics.StateDelta -> *generated.StateDelta
+func stateDeltaToStateDelta(d basics.StateDelta) *generated.StateDelta {
+	if len(d) == 0 {
+		return nil
+	}
+	var delta generated.StateDelta
+	for k, v:= range d {
+		delta = append(delta, generated.EvalDeltaKeyValue{
+			Key:   base64.StdEncoding.EncodeToString([]byte(k)),
+			Value: generated.EvalDelta{
+				Action: uint64(v.Action),
+				Bytes:  strOrNil(base64.StdEncoding.EncodeToString([]byte(v.Bytes))),
+				Uint:   numOrNil(v.Uint),
+			},
+		})
+	}
+	return &delta
+}
+
+func convertToDeltas(txn node.TxnWithStatus) (*[]generated.AccountStateDelta, *generated.StateDelta) {
+	var localStateDelta *[]generated.AccountStateDelta
+	if len(txn.ApplyData.EvalDelta.LocalDeltas) > 0 {
```

Comment:
nit: from code style perspective, it would be clear to use something like:
if len(txn.ApplyData.EvalDelta.LocalDeltas) == 0 {
   localStateDelta, nil
}
also, I'm not sure if the case of the len(..) == 0 should return a nil or empty array. I generally think that it should return an empty array/map rather than nil ( so that we don't need to consider nil down the road ). but it might not be true in this particular case.

Reply:
You might be right, but I'm not sure. It depends on whether the other ApplyData fields are included when they are empty. I think this can be refined later. I created a ticket for now so it doesn't get forgotten:
https://github.com/algorand/go-algorand-internal/issues/679